tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,37 @@
+"""
+karapace - schema authorization tests
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.client import Client
+from karapace.schema_models import SchemaType, ValidatedTypedSchema
+from tests.utils import new_random_name, schema_avro_json
+from urllib.parse import quote
+
+import aiohttp
+
+
+async def test_remote_client_auth(registry_async_client_auth: Client) -> None:
```

Comment:
This tests the cases:
* no auth
* aladdin with WRITE to `cave-`, implicit to have READ also

I'd test also cases:
* aladdin tries to WRITE to `nick-cave-`
* aladdin to READ from `nick-cave-`
* plainuser to READ from `general` and  `cave-`
* plainuser to WRITE to `general` and `cave-`
* admin READ and WRITE

Reply:
`test_sr_list_subjects` now tests with three users to cover this